### ISSUE TITLE
Controller stuck reconciling when it finds an invalid ingress config

### ISSUE THREAD

**Issue by ivanfoo** [kubernetes-sigs/aws-load-balancer-controller#3870]
**Describe the bug**
The controller logs the error and stops reconciling any Ingresses, instead of just ignoring the problematic one  

**Steps to reproduce**
For instance, creating a Service with OIDC authentication annotations that reference a non-existing Secret. All the Ingresses in the same IngressGroup stop being reconciled. 

**Expected outcome**
The controller should log the error but keep processing any other Ingress found.  

**Environment**

* AWS Load Balancer controller version: 2.6.0
* Kubernetes version: 1.28
* Using EKS (yes/no), if so version?: yes, 1.28

**Additional Context**:
Related issues:
- https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/2669
- https://github.com/kubernetes-sigs/aws-load-balancer-controller/issues/2042

**Comment by huangm777** [comment]
Thank you for your report, we will look at this as soon as we can.

**Comment by Sir-Jacques** [comment]
Just out of curiousity as we're hitting a similar issue.

When there is an indirectly invalid Ingress (no valid certificate present in AWS in this case) it locks the entire cluster as no Ingresses can reconcile/finalize anymore. We think finalization must never be impacted by something like this, but it seems closely related to the issues you're experiencing.

Is this deletion-lock happening on your cluster too?

**Comment by ivanfoo** [comment]
@Sir-Jacques yes, exactly like that. Is it impacting all Ingresses within the same IngressGroup or just all Ingresses?

**Comment by Sir-Jacques** [comment]
> @Sir-Jacques yes, exactly like that. Is it impacting all Ingresses within the same IngressGroup or just all Ingresses?

In our setup there is only one ingress-group so I'm not sure what happens to other ingresses that create an ALB (due to infra-restrictions from our AWS admins I cannot test this easily unfortunately).

**Comment by wweiwei-li** [comment]
If you are using ingress group, the behavior is expected. One bad ingress will impact all ingresses within the ingress group.

**Comment by ivanfoo** [comment]
@wweiwei-li well, it's not a recommendation, there is not any other option... but why an invalid ingress has to impact all its neighbors within the same ingress group? Which is the reasoning behind that decision? Wouldn't it make more sense tho just ignore the broken Ingress instead?

**Comment by peterbosalliandercom** [comment]
is there any update on this? We are awaiting this and blocks our production readyness since a noisy neighbour can block all ingresses in the cluster. We have a lot of teams on the cluster. @wweiwei-li @ivanfoo

**Comment by peterbosalliandercom** [comment]
A solution could be that you can make the "alb.ingress.kubernetes.io/group.name" a list of strings where we can link ingressess to certain tenants. This way only the tenants are affected and not the whole cluster.

**Comment by ivanfoo** [comment]
This is critical for us too and probably for anybody sharing the same ALB between different applications @peterbosalliandercom
